Classes/Core/Runtime/FormRuntime.php

```diff
@@ -181,7 +188,12 @@ protected function initializeFormStateFromRequest()
             $this->formState = new FormState();
         } else {
             $serializedFormState = $this->hashService->validateAndStripHmac($serializedFormStateWithHmac);
-            $this->formState = unserialize(base64_decode($serializedFormState), ['allowed_classes' => [FormState::class, \DateTime::class, \DateTimeImmutable::class]]);
+            $allowedClassesInFormState = Arrays::removeEmptyElementsRecursively($this->allowedClassesInFormState);
+            if (is_array($allowedClassesInFormState) && !empty($allowedClassesInFormState)) {
+                $this->formState = unserialize(base64_decode($serializedFormState), ['allowed_classes' => $this->allowedClassesInFormState]);
+            } else {
+                $this->formState = unserialize(base64_decode($serializedFormState));
```

Comment:
This fallback to an unsafe mode seems dangerous to me.
Couldn't we just simplify this to a single line:
```php
$this->formState = unserialize(base64_decode($serializedFormState), ['allowed_classes' => array_filter($this->allowedClassesInFormState)]);
```
?

Reply:
Yes we can. It is  much better solution, thanks.